src/engraving/dom/note.cpp

```diff
@@ -2274,7 +2274,28 @@ int Note::ppitch() const
 
 int Note::epitch() const
 {
-    return m_pitch - (concertPitch() ? 0 : transposition());
+    return m_pitch - (concertPitch() ? 0 : transposition()) + linkedOttavaPitchOffset();
+}
+
+//---------------------------------------------------------
+//   linkedOttavaPitchOffset
+//    if an ottava exists in another part but has been excluded
+//    from this one, this returns the corresponding pitch offset
+//---------------------------------------------------------
+
+int Note::linkedOttavaPitchOffset() const
```

Comment:
Yeah, I liked my idea too (surprise), and if we accepted the change in user interaction it would have taken literally 1 loc. But given that the change in UI was not accepted, working with my idea felt wrong (we'd be basically using a representation model opposite to the UI we want to achieve, and working backwards to achieve it) and it would have required a lot more code than this. Cause it's not just add/remove, it's also the shift + arrow extensions that need change, not to mention dragging. Plus, the changes in add/remove themselves then interfere with exclude/reinclude from parts. Plus (this one independently of the UI) we'd also need to do migration logic for older scores because we're changing the meaning of the ottava marks so old scores would read wrong. 
I can't do that now unfortunately. The good thing about this (slightly hacky) solution is that it is short and self-contained, so we can easily change it in future if we want to.